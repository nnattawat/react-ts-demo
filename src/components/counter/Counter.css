.count-result {
  font-size: 1.3rem;
  padding: 10px 0;
}

.count-controller button:not(:last-child) {
  margin-right: 5px;
}
