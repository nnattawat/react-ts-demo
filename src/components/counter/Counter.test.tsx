import Counter from '../counter';

describe('Counter', () => {

  it('render conrrectly', () => {
    const el = renderer
        .create(<Counter />)
        .toJSON();
    expect(el).toMatchSnapshot();
  });

  it('should call onChange when button is clicked', () => {
    const mockFunction = jest.fn();
    const  el = renderer.create(<Counter onChange={mockFunction} />);
    el.root.findAllByType('button')[0].props.onClick();
    expect(mockFunction).toHaveBeenCalled();
  });
});
