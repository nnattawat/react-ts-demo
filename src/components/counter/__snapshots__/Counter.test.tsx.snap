// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter render conrrectly 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div>
    Count: 
    0
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Reset
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      subtract
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      add
    </span>
  </button>
</div>
`;
